simpeg/utils/mesh_utils.py

```diff
@@ -97,15 +100,192 @@ def surface2inds(vrtx, trgl, mesh, boundaries=True, internal=True):
     return insideGrid
 
 
-################################################
-#             DEPRECATED FUNCTIONS
-################################################
-meshTensor = deprecate_function(
-    unpack_widths, "meshTensor", removal_version="0.19.0", error=True
-)
-closestPoints = deprecate_function(
-    closest_points_index, "closestPoints", removal_version="0.19.0", error=True
-)
-ExtractCoreMesh = deprecate_function(
-    extract_core_mesh, "ExtractCoreMesh", removal_version="0.19.0", error=True
-)
+def _closest_grid_indices(grid, pts, dim=2):
+    """Return indices of closest gridded points for a set of input points.
+
+    Parameters
+    ----------
+    grid : (n, dim) numpy.ndarray
+        A gridded set of points.
+    pts : (m, dim) numpy.ndarray
+        Points being projected to gridded locations.
+    dim : int
+        The `dim` input is deprecated and will be removed in SimPEG v0.24.0.
+
+    Returns
+    -------
+    (n,) numpy.ndarray
+        Indices of the closest gridded points for all *pts* supplied.
+    """
+    if grid.squeeze().ndim == 1:
+        grid_inds = np.asarray(
+            [np.abs(pt - grid).argmin() for pt in pts.tolist()], dtype=int
+        )
+    else:
+        tree = cKDTree(grid)
+        _, grid_inds = tree.query(pts)
+
+    return grid_inds
+
+
+def get_discrete_topography(mesh, active_cells, option="top"):
```

Comment:
I think having this method return a N-1D mesh, and an array in all cases would simplify the `shift_to_discrete_topography` function, and make it so you don't have to reimplement all of the `_closest_grid_indeces` functionality that is already built into discretize.

Reply:
For this project, I intended to preserve the outputs of the original functionality when the original inputs are used. The goals were to:
1. add functionality
2. have the functionality named and located within SimPEG appropriately.

If we make changes to the underlying methods or outputs, then we would be making new functions. Is that what you would like to do? In that case, the original functionality would remain in static_utils.py, but be deprecated.